vignettes/epiparameter.Rmd

```diff
@@ -0,0 +1,521 @@
+
+```{r generate generation time}
+# load data 
+data("Germany.1918")
+
+# generate generation times from discrete gamma distribution with mean 3 and 
+# standard deviation 1.5
+mGT <- generation.time("gamma", c(3, 1.5))
+```
+
+```{r, estimate Rt with R0}
+out <- estimate.R(
+  epid = Germany.1918, 
+  GT = mGT, 
+  begin = 1, 
+  end = 27, 
+  methods = "TD", 
+  nsim = 1e4
+)
+```
+
+```{r plot R0, eval=FALSE}
+plot(out)
+```
+
+```{r plotting function for R0, echo=FALSE, fig.align = "center", fig.height = 8, fig.width = 8, out.width = "500px", fig.cap="Estimated value of $R_t$ for the incidence data of Spanish flu for Germany 1918-1919."}
+
+plot_R0 <- function(out) {
+  plot(
+    as.Date(names(out$estimates$TD$R)), 
+    out$estimates$TD$R, 
+    ylim = c(0, max(out$estimates$TD$conf.int)), 
+    type = "l", 
+    xlab = "Date", 
+    ylab = "Rt"
+  )
+  
+  polygon(
+    x = c(rev(as.Date(rownames(out$estimates$TD$conf.int))), 
+          as.Date(rownames(out$estimates$TD$conf.int))), 
+    y = c(rev(as.vector(out$estimates$TD$conf.int["lower"])$lower),
+          as.vector(out$estimates$TD$conf.int["upper"])$upper), 
+    col = "grey80", 
+    border = NA
+  )
+  
+  lines(as.Date(names(out$estimates$TD$R)), out$estimates$TD$R)
+}
+
+plot_R0(out)
+```
+
+Using the same method to calculate $R_t$ but this time using the data on the outbreak of ebola from Sierra Leone from 2014 and using the a serial interval from `epiparameter` we can again estimate $R_t$.
+
+```{r R0 with epiparameter}
+data("ebola_sierraleone_2014")
+
+ebola_incidence <- incidence(
+  x = ebola_sierraleone_2014, 
+  date_index = "date_of_onset"
+)
+
+# format incidence data for `estimate.R()`
+ebola_incidence_R0 <- ebola_incidence$count
+names(ebola_incidence_R0) <- ebola_incidence$date_index
+
+ebola_si <- epidist(pathogen = "ebola", delay_dist = "serial_interval")
+
+si_mean_sd <- gamma_shapescale2meansd(
+  shape = ebola_si$param[["shape"]], 
+  scale = ebola_si$param[["scale"]]
+)
+
+mGT <- generation.time(type = "gamma", val = c(si_mean_sd$mean, si_mean_sd$sd))
+```
+
+```{r, load R0 data, echo=FALSE}
+load(system.file(
+  "extdata", 
+  "R0_data.rda", 
+  package = "epiparameter", 
+  mustWork = TRUE
+))
+```
+
+```{r, run R0 ebola, eval=FALSE}
+out <- estimate.R(
+  epid = ebola_incidence_R0, 
+  GT = mGT, 
+  begin = 1, 
+  end = 27, 
+  methods = "TD", 
+  nsim = 1000
+)
+```
+
+```{r plot R0 ebola, eval=FALSE}
+plot(out)
+```
+
+```{r plot Rt ebola, echo=FALSE, fig.align = "center", fig.height = 8, fig.width = 8, out.width = "500px", fig.cap="Estimated value of $R_t$ for the incidence data of ebola for Sierra Leone 2014."}
+plot_R0(out)
+```
```

Comment:
````suggestion
```{r plot R0 ebola, fig.align = "center", fig.height = 8, fig.width = 8, out.width = "500px", fig.cap="Estimated value of $R_t$ for the incidence data of ebola for Sierra Leone 2014."}
plot(out$estimates$TD)
```
````